### ISSUE TITLE
TrailingWhitespace hook fails when un-modified lines have trailing whitespace

### ISSUE THREAD

**Issue by derwiki-adroll** [brigade/overcommit#197]
Front-ender reported this error; here is a succinct repro. You'll note `problem_on_unmodified_line: ignore`; I actually have this set in the `ALL` section of my `.overcommit.yml` as well, and the error still presents.

Repro script:
```
#!/bin/bash -e

set -v

# set up repo with trailing whitespace error
mkdir trailing_whitespace && cd trailing_whitespace

echo "apples" > fruits.html
echo "bananas    " >> fruits.html # whitespace error, line 2
echo "oranges" >> fruits.html

git init && git add fruits.html && git commit -m "Initial commit with ws error"

# install overcommit
overcommit --install
echo "PreCommit:" > .overcommit.yml
echo "  TrailingWhitespace:" >> .overcommit.yml
echo "    enabled: true" >> .overcommit.yml
echo "    problem_on_unmodified_line: ignore" >> .overcommit.yml

# add a new line after 3, line 4
echo "kiwi" >> fruits.html
git add fruits.html

# fails because of trailing line space on line 2
git commit -m "Add kiwi to fruits.html"
```

Sample output of script, including failure:
```
$ rm -rf trailing_whitespace/ && bash trailing-whitespace-html-bug.sh

# set up repo with trailing whitespace error
mkdir trailing_whitespace && cd trailing_whitespace

echo "apples" > fruits.html
echo "bananas    " >> fruits.html # whitespace error, line 2
echo "oranges" >> fruits.html

git init && git add fruits.html && git commit -m "Initial commit with ws error"
Initialized empty Git repository in /Users/adamderewecki/projects/trailing_whitespace/.git/
[master (root-commit) e4c3771] Initial commit with ws error
 1 file changed, 3 insertions(+)
 create mode 100644 fruits.html

# install overcommit
overcommit --install
Installing hooks into /Users/adamderewecki/projects/trailing_whitespace
Successfully installed hooks into /Users/adamderewecki/projects/trailing_whitespace
echo "PreCommit:" > .overcommit.yml
echo "  TrailingWhitespace:" >> .overcommit.yml
echo "    enabled: true" >> .overcommit.yml
echo "    problem_on_unmodified_line: ignore" >> .overcommit.yml

# add a new line after 3, line 4
echo "kiwi" >> fruits.html
git add fruits.html

# fails because of trailing line space on line 2
git commit -m "Add kiwi to fruits.html"
Running pre-commit hooks
Checking for trailing whitespace.................[TrailingWhitespace] FAILED
Trailing whitespace detected:
/Users/adamderewecki/projects/trailing_whitespace/fruits.html:2:bananas

✗ One or more pre-commit hooks failed
```

**Comment by derwiki-adroll** [comment]
Reported by @rholmes1

**Comment by sds** [comment]
Thanks for the report, @derwiki-adroll. Nice to see you 'round these parts.

The issue is that `TrailingWhitespace` was implemented in a time way before we supported the special `problem_on_unmodified_line` option, so we never got around to updating it. I've fixed this in 06d5d754376430e74d2631c1107ebb78e4beb146.

**Comment by derwiki-adroll** [comment]
Thanks for fixing so quickly @sds! I've verified that HEAD works as expected.

I'm trying to get the AdRoll team started using overcommit, so you may be hearing more from me in the upcoming months. Off to a fantastic start though, <3 overcommit
